File: spring/src/main/java/com/example/shoppingCart/models/Item.java
package com.example.shoppingCart.models;

import lombok.Data;

import javax.persistence.*;
import java.util.List;

@Entity
@Data
public class Item {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    private String name;
    private double price;

    @OneToMany(mappedBy = "item")
    private List<OrderDetail> orderDetails;

    @OneToMany(mappedBy = "item")
    private List<CartDetail> cartDetails;
}

File: Submission/angular/shopping-cart/src/app/models/Item.ts
import { ICartDetail } from './CartDetail';
import { IOrderDetail } from './OrderDetail';

export interface IItem {
    id: number,
    name: string,
    price: number,
    orderDetails: IOrderDetail[],
    cartDetails: ICartDetail[]
}

export class Item implements IItem {
    id: number = 0;
    name: string;
    price: number;
    orderDetails: IOrderDetail[];
    cartDetails: ICartDetail[];
}